lib/helm/helmutils.go

```diff
@@ -30,10 +30,18 @@ import (
 // vals merges values from files specified via -f/--values and directly via
 // --set or --set-string or --set-file, marshaling them to YAML.
 //
-// This function was copied from Helm:
+// This function was copied from Helm with slight modifications:
 //
 // https://github.com/helm/helm/blob/v2.12.0/cmd/helm/install.go#L363
 func vals(valueFiles valueFiles, values []string, stringValues []string, fileValues []string, CertFile, KeyFile, CAFile string) ([]byte, error) {
+	base, err := merge(valueFiles, values, stringValues, fileValues, CertFile, KeyFile, CAFile)
+	if err != nil {
+		return []byte{}, trace.Wrap(err)
+	}
+	return yaml.Marshal(base)
+}
+
+func merge(valueFiles valueFiles, values []string, stringValues []string, fileValues []string, CertFile, KeyFile, CAFile string) (map[string]interface{}, error) {
```

Comment:
`merge` will silently overwrite longer with shorter (or the other way around), if values have the same prefix (either in two files or in a file and in a value). I would flag this and warn the user.